### PR TITLE
Refactoring changes in Ambassador Pattern

### PR DESCRIPTION
**Pull request by npathai**
Changes
---
  - Updated test cases to use Junit Assert method as compared to assert keyword 
 - Proper testing of `RemoteService` using `RandomProvider` interface. Introduced `RandomProvider` interface so that randomness can be controlled from test cases. 
 - For readability used constant for representing FAILURE

Todos
---

- [x] Update class diagram once refactoring is approved -> no class diagram

### CODE REVIEW

File: ambassador/src/test/java/com/iluwatar/ambassador/ClientTest.java [hunk 1]
```diff
@@ -35,6 +37,6 @@ public void test() {
     Client client = new Client();
     long result = client.useService(10);
 
-    assert result == 100 || result == -1;
+    assertTrue(result == 100 || result == -1);
```
Comment (IAmPramod) [inline ambassador/src/test/java/com/iluwatar/ambassador/ClientTest.java]: We should use `RemoteServiceInterface.FAILURE` here as well. Also, we should create a contstant for 100 as well and use int respective tests.

File: ambassador/src/test/java/com/iluwatar/ambassador/RemoteServiceTest.java [hunk 2]
```diff
@@ -22,16 +22,43 @@
...
+    long result = remoteService.doRemoteFunction(10);
+    assertEquals(RemoteServiceInterface.FAILURE, result);
+  }
+
+  @Test
+  public void testSuccessfulCall() {
+    RemoteService remoteService = new RemoteService(
+        new StaticRandomProvider(0.2));
+    long result = remoteService.doRemoteFunction(10);
+    assertEquals(100, result);
```
Comment (IAmPramod) [inline ambassador/src/test/java/com/iluwatar/ambassador/RemoteServiceTest.java]: We should create a constant for 100.

File: ambassador/src/test/java/com/iluwatar/ambassador/ServiceAmbassadorTest.java [hunk 3]
```diff
@@ -32,6 +34,6 @@
   @Test
   public void test() {
     long result = new ServiceAmbassador().doRemoteFunction(10);
-    assert result == 100 || result == -1;
+    assertTrue(result == 100 || result == RemoteServiceInterface.FAILURE);
```
Comment (IAmPramod) [inline ambassador/src/test/java/com/iluwatar/ambassador/ServiceAmbassadorTest.java]: We should create a constant for 100.

File: ambassador.patch [hunk 4]
```diff
@@ -0,0 +1,205 @@
+diff --git a/ambassador/src/main/java/com/iluwatar/ambassador/RemoteService.java b/ambassador/src/main/java/com/iluwatar/ambassador/RemoteService.java
```
Comment (IAmPramod) [inline ambassador.patch]: I think this is an unintentional file added to commit. We should remove this.

Comment (npathai) [reply ambassador.patch]: My bad. Will do that. Didn't realize. Thanks.

File: ambassador/src/test/java/com/iluwatar/ambassador/ClientTest.java (hunk 1 above)
Comment (npathai) [reply ambassador/src/test/java/com/iluwatar/ambassador/ClientTest.java]: Oh yeah missed here. Done!
100 is calculated at runtime, so I guess we don't really need a constant as such.

**Comment by iluwatar** [comment]
The solution looks good to me as well

**Comment by npathai** [comment]
@iluwatar It doesn't have a class diagram :thinking: So we can merge this PR and I will add a new PR with class diagram if needed.

**Pull request merged**